src/github_runner_manager/openstack_cloud/openstack_runner_manager.py

```diff
@@ -308,17 +312,17 @@ def delete_runner(
             )
             return None
 
-        logger.debug(
-            "Extract metrics from runner before deletion %s %s", instance_id, instance.server_name
-        )
-        extracted_metrics = runner_metrics.extract(
-            metrics_storage_manager=metrics_storage, runners=set([instance.server_name])
-        )
         logger.debug(
             "Metrics extracted, deleting instance %s %s", instance_id, instance.server_name
         )
         self._delete_runner(instance, remove_token)
         logger.debug("Instance deleted successfully %s %s", instance_id, instance.server_name)
+        logger.debug("Extract metrics for runner %s %s", instance_id, instance.server_name)
+        extracted_metrics = runner_metrics.extract(
+            metrics_storage_manager=self._metrics_storage_manager,
+            runners={instance.server_name},
+            include=True,
+        )
```

Comment:
This fixes two issues (also for non-reactive):

- extract metrics after deletion (because deletion will pull the metrics)
- Use include=True, otherwise the effect is the opposite (all metrics are pulled except for this instance).